api/envoy/config/core/v3/command_operator_format.proto

```diff
@@ -0,0 +1,79 @@
+syntax = "proto3";
+
+package envoy.config.core.v3;
+
+import "google/protobuf/struct.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.config.core.v3";
+option java_outer_classname = "CommandOperatorFormatProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Command operator format]
+
+// Configuration to use multiple :ref:`command operators <config_access_log_command_operators>`
+// to generate an new string in either plain text or JSON format.
+message CommandOperatorFormat {
+  oneof format {
+    // Text access log  :ref:`format string<config_access_log_format_strings>`.
+    // Envoy supports :ref:`custom access log formats <config_access_log_format>` as well as a
+    // :ref:`default format <config_access_log_default_format>`.
+    //
+    // .. code-block::
+    //
+    //   text_format: %RESP_BODY%:%RESPONSE_CODE%
+    //
+    // The following plain text will be created:
+    //
+    // .. code-block::
+    //
+    //   upstream connect error:204
+    //
+    string text_format = 1;
+
+    //
+    // Json access log :ref:`format dictionary<config_access_log_format_dictionaries>`. All values
+    // are rendered as strings.
+    //
+    // .. code-block::
+    //
+    //  json_format:
+    //    status: %RESPONSE_CODE%
+    //    message: %RESP_BODY%
+    //
+    // The following JSON object would be created:
+    //
+    // .. code-block:: json
+    //
+    //  {
+    //    "status": "500",
+    //    "message": "My error message"
+    //  }
+    //
+    google.protobuf.Struct json_format = 2;
+
+    // Typed json access log :ref:`format dictionary<config_access_log_format_dictionaries>`. Values are
```

Comment:
If we're defining a new type, do we need raw `json_format` and `typed_json_format`, or just the latter?

Reply:
Good point, I feel the current json_format is a buggy implementation,  we should just implement typed_json_format,  but still call it json_format.

Reply:
For backward compatibility, keep "typed_json_format"

Reply:
.. as well as `json_format` still? I think it's probably OK if this is a conscious choice, it does seem like an opportunity to cleanup some tech debt though and if we get rid of it, we still can bring it back if someone really wants it still.

Reply:
I think the issue here is we need this for back-compat when supporting the old behavior. I suppose it could be an internal implementation detail and hidden from the new API?

Reply:
How about this?
New API will remove "json_format" and rename "typed_json_format" as "json_format",   but in v3, "json_format", "typed_json_format" can still be specified in deprecated access_log_format, it will retain is current behaviors, this will not break existing customers.  But these fields will be removed in v4, and the feature will be removed too.  Since it is a new version, so it is ok to make a breaking change.

Reply:
+1 SGTM, thank you.

/wait

Reply:
Great, thanks @qiwzhang, looks good.